### ISSUE TITLE
mrb_float() doesnt work with MRB_32BIT unless boxing_word.h is included

### ISSUE THREAD

**Issue by sasq64** [mruby/mruby#5595]
If `MRB_32BIT`  is defined `MRB_WORDBOX_NO_FLOAT_TRUNCATE` will also be defined in `boxing_word.h`, and
`mrb_word_boxing_value_float(mrb_value v)` will not be available.
But that only happens if you include `boxing_word.h`,  and so the `mrb_float(o)` macro will not see the TRUNCATE define and
generate a call to the wrong function...

**Comment by sasq64** [comment]
Actually I was just using MRB_INT32 before... is MRB_32BIT a new flag ?

**Comment by dearblue** [comment]
> is MRB_32BIT a new flag ?

No, it is automatically defined when the run-time target is 32-bit (e.g. ARMv7, i386, ...).

It is not expected to define `MRB_32BIT` for 64-bit environments.
It may be desirable to get an error.

**Comment by sasq64** [comment]
I cross compile to Emscripten, where intts are always 32-bit so I want that to be true on all target systems preferably...

**Comment by matz** [comment]
Closing — the `mrb_float()` macro dispatch in boxing_word.h has been restructured. Current state (see include/mruby/boxing_word.h ~end): under `MRB_WORDBOX_NO_INLINE_FLOAT` (the default on MRB_32BIT), `mrb_float(o)` expands to `mrb_rfloat_value(mrb_val_union(o).fp)` (a static inline using `memcpy`); otherwise it expands to `mrb_word_boxing_value_float()`. Both paths are available from just `<mruby/value.h>` via the standard `<mruby.h>` include chain.
